### PR TITLE
Fix deprecation_switch_testnet.py on Parity light client

### LINKED ISSUE #423: 'The execution failed due to an exception' in deprecation_switch_testnet.py
The gas estimater in the Parity node reports an exception during `deprecation_switch_testnet.py`.

```
(env) ~/src/raiden-contracts(master) $ python raiden_contracts/tests/deprecation_switch_testnet.py  --private-key ~/.local/share/io.parity.ethereum/keys/kovan/UTC--2019-01-08T17-09-24Z--fcf0e5e3-7d74-7abc-773e-a2ccab085547
Web3 provider is RPC connection http://127.0.0.1:8545
<snip>
  File "/home/yh/src/raiden-contracts/raiden_contracts/deploy/__main__.py", line 120, in send_deployment_transaction
    private_key=self.private_key,
  File "/home/yh/src/raiden/env/lib/python3.6/site-packages/raiden_libs/private_contract.py", line 26, in transact
    tx_data = self._call.buildTransaction(*args, **kwargs)
  File "/home/yh/src/raiden/env/lib/python3.6/site-packages/web3/utils/decorators.py", line 14, in _wrapper
    return self.method(obj, *args, **kwargs)
  File "/home/yh/src/raiden/env/lib/python3.6/site-packages/web3/contract.py", line 876, in buildTransaction
    return fill_transaction_defaults(self.web3, built_transaction)
  File "cytoolz/functoolz.pyx", line 232, in cytoolz.functoolz.curry.__call__
  File "/home/yh/src/raiden/env/lib/python3.6/site-packages/web3/utils/transactions.py", line 55, in fill_transaction_defaults
    default_val = default_getter(web3, transaction)
  File "/home/yh/src/raiden/env/lib/python3.6/site-packages/web3/utils/transactions.py", line 26, in <lambda>
    'gas': lambda web3, tx: web3.eth.estimateGas(tx),
  File "/home/yh/src/raiden/env/lib/python3.6/site-packages/web3/eth.py", line 304, in estimateGas
    [transaction],
  File "/home/yh/src/raiden/env/lib/python3.6/site-packages/web3/manager.py", line 112, in request_blocking
    raise ValueError(response["error"])
ValueError: {'code': -32016, 'message': 'The execution failed due to an exception.'}
```

### PR DESCRIPTION
**Pull request by pirapira**
This fixes #423.

#347 might be better, but I haven't tried with the Parity fast-sync client.

### CODE REVIEW

File: raiden_contracts/deploy/__main__.py [hunk 1]
```diff
@@ -109,7 +109,10 @@ def deploy(
             f'Deploying {contract_name} txHash={encode_hex(txhash)}, '
             f'contracts version {self.contract_manager.contracts_version}',
         )
-        receipt = check_succesful_tx(self.web3, txhash, self.wait)
+        (receipt, tx) = check_succesful_tx(self.web3, txhash, self.wait)
+        if not receipt['contractAddress']:  # happens with Parity
+            receipt = dict(receipt)
```
Comment (pirapira) [inline raiden_contracts/deploy/__main__.py]: `receipt` is sometimes an immutable `AttributeDict`.  That's why I'm making a copy.

Comment (loredanacirstea) [reply raiden_contracts/deploy/__main__.py]: Ok, then the location is fine.

File: raiden_contracts/utils/transaction.py [hunk 2]
```diff
@@ -1,25 +1,38 @@
...
 
 
 def wait_for_transaction_receipt(web3, txid, timeout=180):
+    receipt = None
     with Timeout(timeout) as time:
-            while not web3.eth.getTransactionReceipt(txid):
+            while not receipt or not receipt['blockNumber']:
+                print('current receipt:')
+                print(receipt)
+                print('')
```
Comment (loredanacirstea) [inline raiden_contracts/utils/transaction.py]: Consider using the `logging` module.

Comment (pirapira) [reply raiden_contracts/utils/transaction.py]: It was a debugging info. Removed from the commit.

**Pull request merged**